### DISCUSSION TITLE
Mengurutkan alternative choices pada selectInput

### DISCUSSION THREAD

**Discussion by oktriy** [teamalgoritma/community#89] [Q&A] [answered]
Saya memasukkan choices dalam selectInput menggunakan unique(nama_kolom), namun ternyata tidak berurutan.
bagaimana cara mengurutkanya?

![pic 3](https://user-images.githubusercontent.com/101442493/167435328-6160551d-88da-4252-beaa-997f4491681c.png)

**Comment by VictorNugraha** [comment, marked as answer]
Halo, @oktriy !

Hal tersebut dapat terjadi karena _selectInput()_ mengikuti urutan levels pada kolom yang menampung ketiga value tersebut. Untuk mengatur ulang levels pada sebuah kolom kita dapat memanfaatkan fungsi _**factor()**_, pada fungsi tersebut terdapat parameter _**levels**_ yang dapat dimanfaatkan untuk mengatur level pada sebuah kolom:

Berikut contoh codenya:

```
df$nama_kolom <- factor(df$nama_kolom, levels = c("2019", "2020", "2021"))
```

Demikian informasi yang dapat kami sampaikan, semoga dapat menjawab pertanyaan anda. Selamat mencoba! :D
